### PR TITLE
Create salary model and update controllers

### PR DESCRIPTION
**Pull request by paulohfev**
https://github.com/codelittinc/backstage-api/issues/47

### CODE REVIEW

File: spec/factories/salaries.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#
+#  index_salaries_on_user_id  (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+FactoryBot.define do
+  factory :salary do
+    yearly_salary { 1.5 }
```
Comment (kaiomagalhaes) [inline spec/factories/salaries.rb:24]: here it should be the yearly salary like 20000

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -35,6 +35,7 @@ def set_user
   def user_params
```
Comment (kaiomagalhaes) [inline app/controllers/users_controller.rb:35]: please update the controllers tests to add this info

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -35,6 +35,7 @@ def set_user
...
     params.require(:user).permit(:active, :contract_type, :email, :first_name, :last_name, :image_url, :seniority,
```
Comment (kaiomagalhaes) [inline app/controllers/users_controller.rb:36]: please update the user views with the salary information

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (paulohfev) [reply app/controllers/users_controller.rb:36]: hey @kaiomagalhaes isn't the user view already showing the info with the changes made in `_user.json.jbuilder`. I also confirmed on the `/users` endpoint, and we're returning the salaries.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (paulohfev) [reply app/controllers/users_controller.rb:35]: hey @kaiomagalhaes in this case, do you want tests for both create and update actions?

**Review by kaiomagalhaes — APPROVED** [review]

**Pull request merged**